### PR TITLE
Ranking system tests

### PR DESCRIPTION
**Pull request by aq6476**
This PR is meant to add several testing suites to frontend components of the ranking System. It also introductions Several Refactors. Here are the additional components made into separate files: 

- ListPlayers.js
- LoadingIndicator.js
- ServerErrorScreen.js

These are refactored from RankingScreen.js 

Testing Suites are added for each of these files. Together it has reached about 90% code coverage.

### CODE REVIEW

File: Carbon/__tests__/navigation/screens/Ranking/RankingScreen.test.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { render, waitFor, act, fireEvent} from '@testing-library/react-native';
+import RankingScreen from '../../../../navigation/screens/Ranking/RankingScreen';
```
Comment (SnellJ2) [inline Carbon/__tests__/navigation/screens/Ranking/RankingScreen.test.js:3]: Nice tests!

File: Carbon/components/LoadingIndicator.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import {Colors} from '../colors/Colors'
+
+/**
+ * 
+ * @param loading - a state variable to determine if loading or not
+ * @returns a null element or a loading screen indicator that will overlay over all existing elements
+ */
+export default function LoadingIndicator({loading}) {
+    if(loading)
+    return (<ActivityIndicator size="large" color={Colors.primary.RAISIN_BLACK} style={LoadingIndicatorStyle} testID="loading-indicator"/>)
```
Comment (SnellJ2) [inline Carbon/components/LoadingIndicator.js:13]: Useful, thanks for adding this as it can be used multiple times.

File: Carbon/navigation/screens/Ranking/RankingScreen.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+import LoadingIndicator from "../../../components/LoadingIndicator";
+import {Colors} from "../../../colors/Colors";
+import ListPlayers from './ListPlayers';
 const PAGE_SIZE = 15;
 
 //Constants - These are to be removed and placed entirely when we build a user session
 
-const API_Entry_RANK_URL = "http://localhost:3000/api/user/rank/"
-const API_Entry_LEADERBOARD_URL = "http://localhost:3000/api/user/leaderboard/"
+const API_Entry_RANK_URL = "http://192.168.0.232:3000/api/user/rank/"
```
Comment (SnellJ2) [inline Carbon/navigation/screens/Ranking/RankingScreen.js:12]: Maybe comment this out so it doesn't cause issues

**Review by SnellJ2 — APPROVED** [review]
I recently had the pleasure of reviewing the Jest test suite written by you and I must say that I am thoroughly impressed by your work. The test suite was well-organized, comprehensive, and covered all the necessary aspects of the codebase.

The person demonstrated a clear understanding of Jest and the best practices for writing tests. The tests were easy to read and understand, and the use of descriptive test names made it simple to identify the purpose of each test.

The test coverage was also impressive. It covered all the important code paths, including error handling, edge cases, and input validation. The person also made excellent use of Jest's powerful features, such as mocking and snapshot testing.

**Review by erickserr125 — APPROVED** [review]
So this PR has a lot of components that I think I can breakdown and I'll start by saying that the tests you wrote are crucial for increasing the code coverage in our project, particularly in the backend as you mentioned in ServerErrorScreen.js. 

In addition, the additions you made to the ranking page is also a welcome change as it will vastly improve the front-end. I approve the changes that are made.
